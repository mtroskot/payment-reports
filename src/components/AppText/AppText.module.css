.TITLE {
  font-family: Roboto;
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 28px;
  margin: 0;
}

.SUBTITLE {
  font-family: Roboto;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 19px;
  color: var(--GRAY_BLUE);
  margin: 0;
}

.LINKS {
  font-family: Roboto;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 19px;
  color: var(--DARK_BLUE);
  margin: 0;
}

.SMALL_TEXT {
  font-family: Roboto;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;
  color: var(--WHITE);
  margin: 0;
}

.lines {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}
