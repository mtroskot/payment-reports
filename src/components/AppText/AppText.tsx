import React, { CSSProperties, HTMLProps, useMemo } from "react";
import styles from "./AppText.module.css";
import { TypographyType } from "src/types/interfaces";
import classNames from "classnames";

interface AppTextProps {
  type: TypographyType;
  children: React.ReactNode | React.ReactNode[];
  color?: string;
  numberOfLines?: number;
  className?: string;
}

const AppText: React.FC<AppTextProps & HTMLProps<HTMLParagraphElement>> = ({
  className,
  color,
  numberOfLines,
  type,
  children,
  ...restProps
}) => {
  const style = useMemo<CSSProperties>(
    () => ({
      color: color,
      lineClamp: numberOfLines,
      WebkitLineClamp: numberOfLines,
    }),
    [color, numberOfLines]
  );

  return (
    <p
      className={classNames(styles[type], className, numberOfLines ? styles.lines : undefined)}
      style={style}
      {...restProps}
    >
      {children}
    </p>
  );
};

export default React.memo(AppText);
