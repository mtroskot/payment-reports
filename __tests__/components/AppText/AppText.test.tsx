import * as React from "react";
import { render, screen } from "@testing-library/react";
import AppText from "src/components/AppText";

describe("renders AppText correctly", () => {
  it("should render default text without any additional styles", async () => {
    const text = "Hello World";
    const { container } = render(<AppText type={"TITLE"}>{text}</AppText>);
    expect(screen.getByText(text)).toBeInTheDocument();
    expect(screen.getByText(text).tagName).toBe("P");
    expect(container).toMatchSnapshot();
  });
  it("should render default text with  additional styles", async () => {
    const text = "Hello World";
    const { container } = render(
      <AppText type={"TITLE"} numberOfLines={2} color={'red'}>
        {text}
      </AppText>
    );
    expect(screen.getByText(text)).toBeInTheDocument();
    expect(screen.getByText(text)).toHaveStyle("line-clamp: 2; color: red");
    expect(container).toMatchSnapshot();
  });
});
