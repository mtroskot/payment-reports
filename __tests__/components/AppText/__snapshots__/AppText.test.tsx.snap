// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AppText correctly should render default text with  additional styles 1`] = `
<div>
  <p
    class=""
    style="color: red; line-clamp: 2;"
  >
    Hello World
  </p>
</div>
`;

exports[`renders AppText correctly should render default text without any additional styles 1`] = `
<div>
  <p
    class=""
  >
    Hello World
  </p>
</div>
`;
